spec/features/hiring_staff_can_visit_the_public_listings_spec.rb

```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+RSpec.feature 'School viewing public listings' do
+  let(:school) { create(:school) }
+  before(:each) do
+    stub_hiring_staff_auth(urn: school.urn)
+  end
+
+  scenario 'A signed in school should see a link back to their own dashboard when viewing public listings' do
+    visit school_path(school)
```

Comment:
This test would be more reliable if it followed what the user would do in their browser. They'd start on the root path and click about

1. Then click the link to sign in as a school
2. Then click the 'School Jobs' title to go back to the public side
3. Then finally click the new My school link
4. And check the page contains something correct.